### PR TITLE
Add future default true to Feature flags

### PR DESCRIPTION
**Pull request by ericholscher**
This also renames `default_true` to `past_default_true`,
making it more explicit as to which does which.

This will give us a lot more flexibility with feature flags,
since we often want to have one of these options with them.

### CODE REVIEW

File: readthedocs/projects/migrations/0064_add_feature_future_default_true.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    dependencies = [
+        ('projects', '0063_extend_domain_from_timestamp_model'),
+    ]
+
+    operations = [
+        migrations.AlterModelOptions(
+            name='environmentvariable',
+            options={'get_latest_by': 'modified', 'ordering': ('-modified', '-created')},
+        ),
```
Comment (humitos) [inline readthedocs/projects/migrations/0064_add_feature_future_default_true.py:16]: This looks unrelated here. I don't know where this came from... :confused:

**Review by humitos — COMMENTED** [review]
I think this is super handy!

I know that @stsewd were doing multiple migrations when deleting fields to avoid breaking on deploy. We may need to do the same here since we are renaming a field, so old instances will try to hit the old name.

**Comment by stsewd** [comment]
yeah, this is going to break during deploy. We should do something like:

During a first deploy

- create the new fields, don't delete the old one (make it nullable here)
- Run a data migration from the data of the old field to the new field (we shouldn't create new feature flags with default=true during this time, or make the data migration during the second deploy)

During a second deploy

- Replace the usage of the old field with the new field
- Delete the old field

**Comment by humitos** [comment]
Once this gets merged and deployed, I'd like to mark `future_default_true=True` to the following feature flags and update its `add_date` accordingly:

* LIMIT_CONCURRENT_BUILDS
* DEDUPLICATE_BUILDS
* SPHINX_PARALLEL

BTW, using the `add_date` (creation date) to compare which projects are going to be affected by the feature flag may end up with "invalid" datetime if it's updated. We are using the same datetime for two different things: "creation date of the object" and "date the feature start taking effect". If we want to keep those different dates, we should add a different field into the model.

**Comment by ericholscher** [comment]
Alright, I updated this PR to not change the existing model. I think it's fine for now, and we can adjust in the future.

**Comment by ericholscher** [comment]
> BTW, using the add_date (creation date) to compare which projects are going to be affected by the feature flag may end up with "invalid" datetime if it's updated. We are using the same datetime for two different things: "creation date of the object" and "date the feature start taking effect". If we want to keep those different dates, we should add a different field into the model.

Agreed. We should probably add a date field and use that, but I don't know if we want it in this PR or not :/

File: readthedocs/projects/models.py [hunk 2]
```diff
@@ -1743,7 +1743,11 @@ def add_features(sender, **kwargs):
         auto_now_add=True,
     )
     default_true = models.BooleanField(
```
Comment (humitos) [inline readthedocs/projects/models.py:1745]:
```suggestion
    # TODO: rename this field to `past_default_true` and follow this steps when deploying
    # https://github.com/readthedocs/readthedocs.org/pull/7524#issuecomment-703663724
    default_true = models.BooleanField(
```

**Review by humitos — APPROVED** [review]

**Review by stsewd — APPROVED** [review]

**Comment by humitos** [comment]
> We should probably add a date field and use that, but I think the `created` date will never be automatically updated, so I don't think it's a huge issue.

It's not because it will be updated automatically, just that when we changed it manually we _loose_ the original date that feature was created. For example, after today's deploy I want every new project to use a feature that already existed, so I will change the `created` date and loose the original one.

**Comment by ericholscher** [comment]
Ah, I see what you mean. Do we care about the actual created date on the model?

**Comment by humitos** [comment]
I don't think it's important, but it's a nice to have to me.
